src/main/java/edu/hm/hafner/analysis/Report.java

```diff
@@ -783,6 +788,88 @@ public int hashCode() {
         result = 31 * result + duplicatesSize;
         return result;
     }
+    
+    private void writeObject(ObjectOutputStream oos) throws IOException {
+        oos.defaultWriteObject();
+        Iterator<Issue> issueIterator = iterator();
+        while(issueIterator.hasNext()) {
+            Issue issue = issueIterator.next();
+            
+            oos.writeUTF(issue.getPath());
+            oos.writeUTF(issue.getFileName());
+            oos.writeInt(issue.getLineStart());
+            oos.writeInt(issue.getLineEnd());
+            oos.writeInt(issue.getColumnStart());
+            oos.writeInt(issue.getColumnEnd());
+            oos.writeObject((LineRangeList) issue.getLineRanges());
+            oos.writeUTF(issue.getCategory());
+            oos.writeUTF(issue.getType());
+            oos.writeUTF(issue.getPackageName());
+            oos.writeUTF(issue.getModuleName());
+            oos.writeObject(issue.getSeverity());
+            oos.writeUTF(issue.getMessage());
+            oos.writeUTF(issue.getDescription());
+            oos.writeUTF(issue.getOrigin());
+            oos.writeUTF(issue.getReference());
+            oos.writeUTF(issue.getFingerprint());
+            oos.writeObject(issue.getAdditionalProperties());
+            oos.writeObject(issue.getId());
+        }
+    }
+    
+    private void readObject(ObjectInputStream ois) throws IOException, ClassNotFoundException {
+        ois.defaultReadObject();
+        elements = new LinkedHashSet<>();
+        TreeStringBuilder tsb = new TreeStringBuilder();
+        while(ois.available() > 0) {
+            
+            final TreeString pathTreeString = tsb.intern(ois.readUTF());
```

Comment:
No `TreeString` instance required here (a lot of additional ones are following below)